image/distributionutil/registry.go

```diff
@@ -35,7 +35,7 @@ func NewV2Repository(named reference.Named, actions ...string) (distribution.Rep
 	}
 
 	authConfig, authErr := utils.GetDockerAuthInfoFromDocker(named.Domain())
-	if err != nil {
+	if authErr != nil {
 		logger.Warn("failed to get auth info, err: %s", authErr)
 		return nil, err
```

Comment:
I am wondering if we need return authErr here.

Reply:
> I am wondering if we need return authErr here.

For example, when pulling a public image fails and there is no login, I think it is necessary to return a pull failure error rather than an authentication error

Reply:
Delete the warn log, or using debug log level.